File: rust/src/models/get_checkout_session_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::checkout_session::CheckoutSession;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct GetCheckoutSessionResponse {
    pub session: CheckoutSession,
}

impl GetCheckoutSessionResponse {
    pub fn new(session: CheckoutSession) -> Self {
        Self { session }
    }
}

File: typescript/src/models/getCheckoutSessionResponse.ts
// this file is @generated
import { type CheckoutSession, CheckoutSessionSerializer } from "./checkoutSession";

export interface GetCheckoutSessionResponse {
  session: CheckoutSession;
}

export const GetCheckoutSessionResponseSerializer = {
  _fromJsonObject(object: any): GetCheckoutSessionResponse {
    return {
      session: CheckoutSessionSerializer._fromJsonObject(object["session"]),
    };
  },

  _toJsonObject(self: GetCheckoutSessionResponse): any {
    return {
      session: CheckoutSessionSerializer._toJsonObject(self.session),
    };
  },
};
